geode-core/src/test/java/org/apache/geode/internal/cache/tier/sockets/CacheClientProxyTest.java

```diff
@@ -75,32 +79,41 @@ public void before() throws Exception {
     statsFactory = mock(StatisticsFactory.class);
     proxyStatsFactory = mock(CacheClientProxyStatsFactory.class);
     dispatcherFactory = mock(MessageDispatcherFactory.class);
+    clientUserAuths = mock(ClientUserAuths.class);
+
+    proxyWithSingleUser =
+        new CacheClientProxy(cache, notifier, socket, id, true, (byte) 1, version, 1L, true,
+            securityService, subject, clock, statsFactory, proxyStatsFactory, dispatcherFactory,
+            clientUserAuths);
+
+    proxyWithMultiUser =
+        new CacheClientProxy(cache, notifier, socket, id, true, (byte) 1, version, 1L, true,
+            securityService, null, clock, statsFactory, proxyStatsFactory, dispatcherFactory,
+            clientUserAuths);
   }
 
   @Test
   public void noExceptionWhenGettingSubjectForCQWhenSubjectIsNotNull() {
-    proxy = spy(new CacheClientProxy(cache, notifier, socket, id, true, (byte) 1, version, 1L, true,
-        securityService, subject, clock, statsFactory, proxyStatsFactory, dispatcherFactory));
-    proxy.getSubject("cq");
+    proxyWithSingleUser.getSubject("cq");
   }
 
   @Test
   public void noExceptionWhenGettingSubjectForCQWhenSubjectIsNull() {
-    proxy = spy(new CacheClientProxy(cache, notifier, socket, id, true, (byte) 1, version, 1L, true,
-        securityService, null, clock, statsFactory, proxyStatsFactory, dispatcherFactory));
-    proxy.getSubject("cq");
+    proxyWithMultiUser.getSubject("cq");
   }
 
   @Test
   public void deliverMessageWhenSubjectIsNotNull() {
     when(proxyStatsFactory.create(any(), any(), any()))
         .thenReturn(mock(CacheClientProxyStats.class));
```

Comment:
If this mocking is moved to the `@Before` method then you can avoid having to create a new `CacheClientProxy` on the following lines. This also applies to the `deliverMessageWhenSubjectIsNull()` test.

Reply:
done